### PR TITLE
Use 8.6 reference for the links in the documentation

### PR DESCRIPTION
**Pull request by praveen-elastic**
The documentation for connectors provided in DEVELOPING.md should contain 8.6 branch reference in place of main.

All the following links that contain `/blob/main` will now point to `/blob/8.6` branches:

- the connector client links
- the links to file types eligible for extraction
- the ruby repository links

cc @danajuratoni 

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Considered corresponding documentation changes

### CODE REVIEW

File: docs/DEVELOPING.md [hunk 1]
```diff
@@ -161,7 +161,7 @@ Whether the connector should extract file content from Amazon S3. Default value
 
 #### Content extraction
 
-The connector uses the Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using the Apache text extraction library Tika. Supported file types eligible for extraction can be found [here](https://github.com/elastic/connectors-python/blob/9cf07a96288dcd542b641c51533ee2d427ef56ae/connectors/utils.py#L27).
+The connector uses the Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using the Apache text extraction library Tika. Supported file types eligible for extraction can be found [here](https://github.com/elastic/connectors-python/blob/2e773285035f0c4f2ad1011a4f96ddac1a5d7a77/connectors/sources/s3.py#L25).
```
Comment (danajuratoni) [inline docs/DEVELOPING.md:164]: Same as above

**Comment by danajuratoni** [comment]
Don't forget to backport to v8.6.0

**Review by danajuratoni — APPROVED** [review]
LGTM

**Comment by tarekziade** [comment]
buildkite test this

**Comment by tarekziade** [comment]
buildkite test this

**Comment by tarekziade** [comment]
buildkite test this

**Pull request merged**